File: LeetCode/Study_Plan/DataStructure/C_Plus_Plus/121.cpp
//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution
{
    public:
        int maxProfit(vector<int>& prices)
        {
            int min_price = INT_MAX;
            int max_profit = 0;
            for ( int i = 0; i < prices.size(); i++ )
            {
                if ( min_price > prices[i] )
                {
                    min_price = prices[i];
                }
                if ( prices[i] - min_price  > max_profit )
                {
                    max_profit = prices[i] - min_price;
                }
            }
            return max_profit;
        }
};

File: LeetCode/Study_Plan/DataStructure/Python/121.py
'''https://leetcode.com/problems/best-time-to-buy-and-sell-stock/'''

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = 10 ** 4
        maxProfit = 0
        for i in range(len(prices)):
            if minPrice > prices[i]:
                minPrice = prices[i]
            elif prices[i] - minPrice > maxProfit:
                maxProfit = prices[i] - minPrice
        return maxProfit